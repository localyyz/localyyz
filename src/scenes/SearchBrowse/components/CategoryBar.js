import React from "react";
import { View, StyleSheet, FlatList, TouchableOpacity } from "react-native";

// third party
import { observable, action } from "mobx";
import { Provider, observer, inject } from "mobx-react/native";

// custom
import { Colours, Sizes } from "localyyz/constants";

// local
import CategoryButton from "./CategoryButton";

@observer
export default class CategoryBar extends React.Component {
  @observable selectedCategory = "";

  constructor(props) {
    super(props);

    // bindings
    this.renderItem = this.renderItem.bind(this);
  }

  @action
  onChangeCategory({ title, id }) {
    this.selectedCategory = id;
    this.props.onChangeCategory
      && this.props.onChangeCategory({ title: title, id: id });
  }

  renderItem({ item }) {
    return (
      <CategoryBarItem
        {...item}
        onChangeCategory={id => this.onChangeCategory(id)}/>
    );
  }

  render() {
    const categories = this.props.store.categories
      .filter(cat => cat.id === this.props.id)
      .map(cat => cat.values)[0];

    return (
      <Provider categoryComponent={this}>
        <View style={styles.container}>
          {categories && categories.length > 0 ? (
            <FlatList
              horizontal
              showsHorizontalScrollIndicator={false}
              data={categories}
              renderItem={this.renderItem}
              contentContainerStyle={styles.list}
              keyExtractor={cat => cat.id}/>
          ) : null}
        </View>
      </Provider>
    );
  }
}

@inject((stores, props) => ({
  isSelected: stores.categoryComponent.selectedCategory === props.id
}))
@observer
class CategoryBarItem extends React.Component {
  constructor(props) {
    super(props);

    // bindings
    this.onChange = this.onChange.bind(this);
  }

  onChange() {
    return (
      this.props.onChangeCategory && this.props.onChangeCategory(this.props)
    );
  }

  render() {
    return (
      <TouchableOpacity onPress={this.onChange}>
        <CategoryButton isSmall {...this.props} />
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colours.Foreground
  },

  list: {
    marginBottom: Sizes.InnerFrame / 2,
    paddingHorizontal: Sizes.InnerFrame
  }
});
