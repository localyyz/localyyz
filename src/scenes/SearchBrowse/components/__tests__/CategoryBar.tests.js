import React from "react";
import Renderer from "react-test-renderer";

import CategoryBar, { CategoryBarItem } from "../CategoryBar";

const CATEGORIES = [
  {
    id: "apparel",
    type: "apparel",
    title: "Apparel",
    values: [
      {
        id: "activewear",
        type: "activewear",
        title: "Activewear",
        values: [],
        imageUrl:
          "https://cdn.shopify.com/s/files/1/1825/3189/products/1589d1c85bfdd70a943bbcc2c75374d6.jpg"
      }
    ]
  }
];

const DEFAULT_PROPS = {
  id: "apparel",
  store: {
    categories: CATEGORIES
  }
};

describe("CategoryBar", () => {
  it("should render properly", () => {
    let render = Renderer.create(<CategoryBar {...DEFAULT_PROPS} />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});

describe("CategoryBarItem", () => {
  it("should render properly", () => {
    let render = Renderer.create(
      <CategoryBarItem.wrappedComponent {...CATEGORIES[0].values[0]} />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
