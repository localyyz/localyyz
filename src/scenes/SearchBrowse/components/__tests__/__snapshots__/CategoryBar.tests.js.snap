// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryBar should render properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "marginBottom": 7.5,
        "paddingHorizontal": 15,
      }
    }
    data={
      Array [
        Object {
          "id": "activewear",
          "imageUrl": "https://cdn.shopify.com/s/files/1/1825/3189/products/1589d1c85bfdd70a943bbcc2c75374d6.jpg",
          "title": "Activewear",
          "type": "activewear",
          "values": Array [],
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#000000",
                  "borderRadius": 10,
                  "height": 300,
                  "marginRight": 7.5,
                  "overflow": "hidden",
                  "width": 300,
                },
                Object {
                  "height": 125,
                  "marginRight": 3.75,
                },
              ]
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "uri": "https://cdn.shopify.com/s/files/1/1825/3189/products/1589d1c85bfdd70a943bbcc2c75374d6.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0, 0, 0, 0.2)",
                  "borderRadius": 10,
                  "flex": 1,
                  "justifyContent": "space-between",
                  "overflow": "hidden",
                  "padding": 15,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "borderBottomColor": "#FFFFFF",
                    "color": "#FFFFFF",
                    "fontSize": 22,
                    "fontWeight": "600",
                    "textAlign": "center",
                  }
                }
              >
                Activewear
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`CategoryBarItem should render properly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 10,
          "height": 300,
          "marginRight": 7.5,
          "overflow": "hidden",
          "width": 300,
        },
        Object {
          "height": 125,
          "marginRight": 3.75,
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://cdn.shopify.com/s/files/1/1825/3189/products/1589d1c85bfdd70a943bbcc2c75374d6.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "borderRadius": 10,
          "flex": 1,
          "justifyContent": "space-between",
          "overflow": "hidden",
          "padding": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "borderBottomColor": "#FFFFFF",
            "color": "#FFFFFF",
            "fontSize": 22,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Activewear
      </Text>
    </View>
  </View>
</View>
`;
